core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -174,11 +186,19 @@ public void run() {
             return;
           }
           // Each address forms an EAG
-          ArrayList<EquivalentAddressGroup> servers = new ArrayList<EquivalentAddressGroup>();
+          List<EquivalentAddressGroup> servers = new ArrayList<EquivalentAddressGroup>();
           for (InetAddress inetAddr : resolvedInetAddrs.addresses) {
             servers.add(new EquivalentAddressGroup(new InetSocketAddress(inetAddr, port)));
           }
-          savedListener.onAddresses(servers, Attributes.EMPTY);
+          servers.addAll(resolvedInetAddrs.srvRecords);
+
+          Attributes.Builder attrs = Attributes.newBuilder();
+          if (!resolvedInetAddrs.txtRecords.isEmpty()) {
+            attrs.set(
+                GrpcAttributes.NAME_RESOLVER_ATTR_TXT,
```

Comment:
TXT record is DNS-specific. Let's add "DNS" to the name of this attribute.

Reply:
Done.